presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -1223,29 +1223,60 @@ core::PlanNodePtr VeloxQueryPlanConverterBase::toVeloxQueryPlan(
         sourceNodes[i]);
   }
 
-  if (isHashPartition(node)) {
+  if (type == core::LocalPartitionNode::Type::kGather) {
+    return core::LocalPartitionNode::gather(node->id, std::move(sourceNodes));
+  }
+
+  auto connectorHandle =
```

Comment:
This method is very long. Consider extracting some helper functions.

Reply:
Update!